src/Symfony/Component/HttpFoundation/IpUtils.php

```diff
@@ -153,4 +153,12 @@ public static function checkIp6($requestIp, $ip)
 
         return self::$checkedIps[$cacheKey] = true;
     }
+
+    /**
+     * Anonymizes an IP/IPv6 by removing its last bit
```

Comment:
Shouldn't "bit" be "byte" here?

Reply:
Maybe, would like to first figure out what to do with IPv6s

Reply:
Maybe we could use an alternative anonymizing method based on IP masks:

```php
public static function anonymize(string $ip): string
{
    $packedAddress = inet_pton($ip);
    $anonymizingMask = 4 === strlen($packedAddress) ? '255.255.255.0' : 'ffff:ffff:ffff:ffff:0000:0000:0000:0000';

    return inet_ntop($packedAddress & inet_pton($anonymizingMask));
}
```

Code adapted from https://github.com/geertw/php-ip-anonymizer/blob/master/src/IpAnonymizer.php (MIT license)

Reply:
this comment is still true today. Even for IPv4, it removes a byte, not a bit

Reply:
I think so :)